ftp_s3/S3DownloadOnlyFileSystem.py

```diff
@@ -0,0 +1,30 @@
+from .S3FileSystem import S3FileSystem
+from . import settings
+from pyftpdlib.filesystems import FilesystemError
+
+class S3DownloadOnlyFileSystem(S3FileSystem):
+    def listdir(self, path):
+        return []
+
+    def isfile(self, path):
+        return True
```

Comment:
Thanks @alaayameen, this modification is meant to overwrite the default settings and only serve files, directories unsupported.